docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -1,140 +1,159 @@
-
-and update the `Users `entry with `Users.emailId = UsersContact.Id` and `Users.passphraseEncryptionType = 1`
-
-After this one-time migration the column `Users.email` can be deleted.
-
-### Adaption of BusinessLogic
-
-The following logic or business processes has to be adapted for introducing the Gradido-ID
-
-#### Read-Write Access of Users-Table especially Email
-
-The ORM mapping has to be adapted to the changed and new database schema.
-
-#### Registration Process
-
-The logic of the registration process has to be adapted by
-
-* initializing the `Users.userID` with a unique UUID
-* creating a new `UsersContact `entry with the given email address and *maincontact* as `usedChannel `
-* set `emailID `in the `Users `table as foreign key to the new `UsersContact `entry
-* set `Users.passphraseEncrpytionType = 2` and encrypt the passphrase with the `Users.userID` instead of the `UsersContact.email`
-
-#### Login Process
-
-The logic of the login process has to be adapted by
-
-* search the users data by reading the `Users `and the `UsersContact` table with the email (or alias as soon as the user can maintain his profil with an alias)   as input
-* depending on the `Users.passphraseEncryptionType` decrypt the stored password
-  * = 1 :  with the email
-  * = 2 : with the userID
-
-#### Password En/Decryption
-
-The logic of the password en/decryption has to be adapted by encapsulate the logic to be controlled with an input parameter. The input parameter can be the email or the userID.
-
-#### Change Password Process
-
-The logic of change password has to be adapted by
-
-* if the `Users.passphraseEncryptionType` = 1, then
-
-  * read the users email address from the `UsersContact `table
-  * give the email address as input for the password decryption of the existing password
-  * use the `Users.userID` as input for the password encryption fo the new password
-  * change the `Users.passphraseEnrycptionType` to the new value =2
-* if the `Users.passphraseEncryptionType` = 2, then
-
-  * give the `Users.userID` as input for the password decryption of the existing password
-  * use the `Users.userID` as input for the password encryption fo the new password
-
-#### Search- and Access Logic
-
-A new logic has to be introduced to search the user identity per different input values. That means searching the user data must be possible by
-
-* searching per email (only with maincontact as contactchannel)
-* searching per userID
-* searching per alias
-
-#### Identity-Mapping
-
-A new mapping logic will be necessary to allow using unmigrated APIs like GDT-servers api. So it must be possible to give this identity-mapping logic the following input to get the respective output:
-
-* email -> userID
-* email -> alias
-* userID -> email
-* userID -> alias
-* alias -> email
-* alias -> userID
-
-#### GDT-Access
-
-To use the GDT-servers api the used identifier for GDT has to be switch from email to userID.
+# Introduction of Gradido-ID

```

Comment:
Are there any relevant changes? Please fix end of lines (NL instead of NL CR)

Reply:
done